### PR TITLE
doxygen: 1.8.6 -> 1.8.11

### PR DESCRIPTION
**Pull request by jraygauthier**
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

As per upstream change, now uses cmake.

Tested proper function of both `doxygen` and `doxygen_gui`.

### CODE REVIEW

File: pkgs/development/tools/documentation/doxygen/default.nix [hunk 1]
```diff
@@ -1,45 +1,24 @@
-{ stdenv, fetchurl, perl, python, flex, bison, qt4, CoreServices, libiconv }:
+{ stdenv, cmake, fetchurl, perl, python, flex, bison, qt4, CoreServices, libiconv }:
 
 let
-  name = "doxygen-1.8.6";
+  name = "doxygen-1.8.11";
 in
 stdenv.mkDerivation {
   inherit name;
```
Comment (hrdinka) [inline pkgs/development/tools/documentation/doxygen/default.nix]: Please remove the unnecessary `let ...` from above and define the name directly here. It's now common to use recursive attribute sets. If you add `rec` after `mkDerivation` `name` can be referenced within this derivation (like in `src` below).

File: pkgs/development/tools/documentation/doxygen/default.nix [hunk 2]
```diff
@@ -1,45 +1,24 @@
...
 
-  prePatch = ''
-    substituteInPlace configure --replace /usr/bin/install $(type -P install)
-  '';
-
-  patches = [ ./tmake.patch ];
-
   buildInputs =
-    [ perl python flex bison ]
+    [ cmake perl python flex bison ]
```
Comment (hrdinka) [inline pkgs/development/tools/documentation/doxygen/default.nix]: `cmake` is used to build the package and not as a dependency by the package. It therefor belongs to `nativeBuildInputs`.

**Comment by hrdinka** [comment]
If just noticed that this derivation still is accompanied by the 7 year old `tmake.patch` patch. Please be so kind to remove this as well.

**Comment by jraygauthier** [comment]
@hrdinka: Updated with suggested changes.

**Comment by hrdinka** [comment]
Builds & works for me :+1:

Doxygens maintainer @peti hasn't been mentioned yet (dare you @mention-bot).

**Pull request merged**